### ISSUE TITLE
Metric: SR-SIM

### ISSUE THREAD

**Issue by leihuayi** [photosynthesis-team/piq#188]
**Is your feature request related to a problem? Please describe.**
Spectral Residual based Similarity (SR-SIM) index between two images is the combination of saliency maps made on luminance channel of image and gradient maps (like in FSIM).
*Performance* : Seems to perform slightly better than IW-SSIM according to own paper and HaarPSI paper, and the following studies:
- [A comprehensive evaluation of full-reference image quality assessment algorithms on KADID-10k](https://arxiv.org/pdf/1907.02096), 2019
- [Full-Reference Quality Metric Based on Neural Network to Assess the Visual Quality of Remote Sensing Images](https://www.mdpi.com/2072-4292/12/15/2349/htm#B90-remotesensing-12-02349), 2020

**Describe the solution you'd like**
A pytorch implementation of SR-SIM

**Describe alternatives you've considered**
I have coded a python version out of the matlab code (very similar, except that matlab resize and cv2 resize do not work exactly the same). However I feel it would be more useful to the community if it were part of your repo which is very complete and functional.
I am not very knowledgeable in Pytorch but if you do not have time I can try to implement this feature as contributor.

**Additional context**
Original paper : https://sse.tongji.edu.cn/linzhang/ICIP12/ICIP-SR-SIM.pdf
Authors matlab code : https://github.com/Netflix/vmaf/blob/master/matlab/strred/SR_SIM.m
Python code : https://github.com/leihuayi/FRIQAMetrics/blob/master/friqametrics/srsim.py
